quickwit-search/src/leaf.rs

```diff
@@ -21,28 +21,38 @@
 use crate::collector::{make_collector_for_split, make_merge_collector, GenericQuickwitCollector};
 use crate::SearchError;
 use anyhow::Context;
+use bytes::Bytes;
 use futures::future::try_join_all;
 use itertools::{Either, Itertools};
+use lru::LruCache;
 use quickwit_directories::{CachingDirectory, HotDirectory, StorageDirectory};
 use quickwit_index_config::IndexConfig;
 use quickwit_proto::{LeafSearchResult, SearchRequest, SplitIdAndFooterOffsets, SplitSearchError};
 use quickwit_storage::{BundleStorage, Storage};
 use std::collections::BTreeMap;
 use std::convert::TryInto;
 use std::path::PathBuf;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 use tantivy::{collector::Collector, query::Query, Index, ReloadPolicy, Searcher, Term};
 use tokio::task::spawn_blocking;
 
-/// Opens a `tantivy::Index` for the given split.
-///
-/// The resulting index uses a dynamic and a static cache.
-pub(crate) async fn open_index(
+lazy_static! {
+    static ref SPLIT_FOOTER_CACHE: Mutex<LruCache<String, Bytes>> = Mutex::new(LruCache::new(500));
```

Comment:
We have code to have an LruCache for which the limit is expressed in bytes. It is clunky, but can we use that?

Reply:
I need to make it more generic first, but yes